### ISSUE TITLE
Missing declarations for two variables in the member function “PartitionPage::updateButtons”

### ISSUE THREAD

**Issue by elfring** [calamares/calamares#1766]
I cloned the current development repository once more a moment ago.
I configured then “RelWithDebInfo” as the CMake build type.

Now I am wondering about the following error messages.

```
[Markus_Elfring@fedora Bau]$ LANG=C ninja
[8/162] Building CXX object src/modules/partition/CMakeFiles/calamares_viewmodule_partition.dir/gui/PartitionPage.cpp.o
FAILED: src/modules/partition/CMakeFiles/calamares_viewmodule_partition.dir/gui/PartitionPage.cpp.o 
/usr/lib64/ccache/c++ -DKCOREADDONS_LIB -DPLUGINDLLEXPORT_PRO -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_SHARED -DQT_SHAREDPOINTER_TRACK_POINTERS -DQT_STRICT_ITERATORS -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DWITH_KPMCORE331API -DWITH_KPMCORE42API -DWITH_KPMCORE4API -Dcalamares_viewmodule_partition_EXPORTS -Isrc/modules/partition/calamares_viewmodule_partition_autogen/include -I/home/Markus_Elfring/Test/calamares/Probe2/CX/src/libcalamares -Isrc/libcalamares -I/home/Markus_Elfring/Test/calamares/Probe2/CX/src/libcalamaresui -I/home/Markus_Elfring/Test/calamares/Probe2/CX -I/usr/include/kpmcore -Isrc/libcalamaresui -I/home/Markus_Elfring/Test/calamares/Probe2/CX/src/modules/partition -Isrc/modules/partition -I/usr/include/libcalamares -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/KF5/KCoreAddons -isystem /usr/include/KF5 -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtXml -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtSvg -isystem /usr/include/qt5/QtQuickWidgets -isystem /usr/include/qt5/QtQuick -isystem /usr/include/qt5/QtQmlModels -isystem /usr/include/qt5/QtQml -Wall -Werror=return-type -Wnon-virtual-dtor -Woverloaded-virtual -fdiagnostics-color=auto -O2 -g -fPIC -fPIC -std=gnu++17 -MD -MT src/modules/partition/CMakeFiles/calamares_viewmodule_partition.dir/gui/PartitionPage.cpp.o -MF src/modules/partition/CMakeFiles/calamares_viewmodule_partition.dir/gui/PartitionPage.cpp.o.d -o src/modules/partition/CMakeFiles/calamares_viewmodule_partition.dir/gui/PartitionPage.cpp.o -c /home/Markus_Elfring/Test/calamares/Probe2/CX/src/modules/partition/gui/PartitionPage.cpp
/home/Markus_Elfring/Test/calamares/Probe2/CX/src/modules/partition/gui/PartitionPage.cpp: In member function 'void PartitionPage::updateButtons()':
/home/Markus_Elfring/Test/calamares/Probe2/CX/src/modules/partition/gui/PartitionPage.cpp:186:17: error: 'createTable' was not declared in this scope
  186 |                 createTable = false;
      |                 ^~~~~~~~~~~
/home/Markus_Elfring/Test/calamares/Probe2/CX/src/modules/partition/gui/PartitionPage.cpp:187:17: error: invalid use of member function 'void QWidget::create(WId, bool, bool)' (did you forget the '()' ?)
  187 |                 create = false;
      |                 ^~~~~~
[13/162] Building CXX object src/modules/partition/tests/CMakeFiles/clearmountsjobtests.dir/ClearMountsJobTests.cpp.o
ninja: build stopped: subcommand failed.
```

How should the corresponding variable declarations be restored? :thinking:

**Comment by elfring** [comment]
Would another variable renaming make sense? :thinking:

```diff
diff --git a/src/modules/partition/gui/PartitionPage.cpp b/src/modules/partition/gui/PartitionPage.cpp
index aa05c3a26..d22f6f01d 100644
--- a/src/modules/partition/gui/PartitionPage.cpp
+++ b/src/modules/partition/gui/PartitionPage.cpp
@@ -183,8 +183,8 @@ PartitionPage::updateButtons()
             if ( device->type() == Device::Type::SoftwareRAID_Device
                  && static_cast< SoftwareRAID* >( device )->status() == SoftwareRAID::Status::Inactive )
             {
-                createTable = false;
-                create = false;
+                allow_create_table = false;
+                allow_create = false;
             }
 #endif
         }
```


Was [the commit “[partition] Improve constness, naming”](https://github.com/calamares/calamares/commit/46ca4d93e76abd8ee8b5518b291dc170ca1d3063 "Software update from 2021-08-26") still questionable?

**Comment by demmm** [comment]
Are you building from latest git (pretty sure that is a yes)?  Did this commit cause your build fail?
https://github.com/calamares/calamares/commit/46ca4d93e76abd8ee8b5518b291dc170ca1d3063

**Comment by elfring** [comment]
>Are you building from latest git (…)?

Yes.


>Did this commit cause your build fail?

I got the impression that the preprocessor symbol “`WITH_KPMCORE4API`” was overlooked with the last software update.

**Comment by adriaandegroot** [comment]
Fixed via #1765
